tla-io/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala

```diff
@@ -83,18 +83,19 @@ class SanyParserPassImpl @Inject() (
 
         // write parser output to specified destination, if requested
         options.get[String]("parser", "output").foreach { output =>
-          val file = new File(output)
+          val outfile = new File(output)
+          val outfileName = outfile.toString()
 
-          if (filename.toLowerCase.endsWith(".tla")) {
-            val moduleName = filename.substring(0, filename.length - ".tla".length)
+          if (outfileName.toLowerCase.endsWith(".tla")) {
+            val moduleName = outfileName.substring(0, outfileName.length - ".tla".length)
             writerFactory.writeModuleToTla(rootModule.get.copy(name = moduleName), TlaWriter.STANDARD_MODULES,
-                Some(file))
-          } else if (filename.toLowerCase.endsWith(".json")) {
-            val moduleName = filename.substring(0, filename.length - ".json".length)
+                Some(outfile))
+          } else if (outfileName.toLowerCase.endsWith(".json")) {
+            val moduleName = outfileName.substring(0, outfileName.length - ".json".length)
             writerFactory.writeModuleToJson(rootModule.get.copy(name = moduleName), TlaWriter.STANDARD_MODULES,
```

Comment:
The duplicate logic in these bits makes this kind of problem a bit more likely, but I'll be refactoring this as part of #1077